scala/src/main/scala/smile/clustering/distance/BinaryDistance.scala

```diff
@@ -0,0 +1,11 @@
+package smile.math.distance
+
+/**
+ * @author Beck Gaël
+ *
+ **/
+trait BinaryDistance extends Distance {
```

Comment:
This trait seems redundant to me. It doesn't really add new functionality or (real) constraints to Distance. Do we really need it?